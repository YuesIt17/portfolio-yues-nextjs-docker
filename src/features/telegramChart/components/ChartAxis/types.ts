import {TChartDataLabel} from '@/utils/types';

export type TChartAxis = {
  points: string;
  typeAxis: 'x' | 'y';
  labels: TChartDataLabel[];
  labelPaddingX?: number;
  labelPaddingY?: number;
  maxDataX?: number;
};
