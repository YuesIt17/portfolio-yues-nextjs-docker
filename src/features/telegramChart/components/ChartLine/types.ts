export type TChartLine = {
  points: string;
  fill?: string;
  stroke?: string;
  isVisible?: boolean;
};
