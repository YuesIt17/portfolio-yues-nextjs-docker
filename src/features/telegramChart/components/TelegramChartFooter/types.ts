import {TChartDataLine} from '@/utils/types';

export type TTelegramChartFooter = {
  data: TChartDataLine[] | undefined;
  onChange: (line: TChartDataLine, isVisible: boolean) => void;
};
