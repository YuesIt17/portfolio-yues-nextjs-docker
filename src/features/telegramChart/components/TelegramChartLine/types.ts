import {TChartDataLine, TChartPrepareData} from '@/utils/types';

export type TTelegramChartLine = {
  data: TChartDataLine[];
  width?: number;
  peddingSize?: number;
} & Omit<TChartPrepareData, 'lines'>;
