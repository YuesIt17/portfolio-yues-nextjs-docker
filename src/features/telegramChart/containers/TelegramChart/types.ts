export type TTelegramChart = {
  width: number;
};

export type TTelegramChartStyle = TTelegramChart;
