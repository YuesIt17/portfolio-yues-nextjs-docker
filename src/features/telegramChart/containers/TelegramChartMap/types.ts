import {TChartDataLine} from '@/utils/types';

export type TTelegramChartMap = {
  width: number;
  data: TChartDataLine[];
  maxDataX: number;
  maxDataY: number;
};

export type TTelegramChartMapStyle = {
  width: number;
};
