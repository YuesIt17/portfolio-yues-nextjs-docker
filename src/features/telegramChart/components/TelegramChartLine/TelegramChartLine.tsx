import {TChartDataLine} from '@/utils/types';
import React, {memo, useContext} from 'react';
import {ClassComponentContext} from '../../pages/TelegramChartPage';
import {ChartLineClass} from '../../sampleClassComponent/components';
import {ChartAxis} from '../ChartAxis';
import {ChartLine} from '../ChartLine';
import {TTelegramChartLine} from './types';

const Line = memo(
  ({
    hasClassComponent,
    line,
  }: {
    hasClassComponent: boolean;
    line: TChartDataLine;
  }) => (
    <>
      {hasClassComponent ? (
        <ChartLineClass
          points={line.points}
          stroke={line.color}
          isVisible={line.isVisible}
        />
      ) : (
        <ChartLine
          points={line.points}
          stroke={line.color}
          isVisible={line.isVisible}
        />
      )}
    </>
  )
);
Line.displayName = 'Line';

export const TelegramChartLine = ({
  data,
  maxDataX,
  maxDataY,
  labelsX,
  labelsY,
  width,
  peddingSize = 0,
}: TTelegramChartLine) => {
  const {hasClassComponent} = useContext(ClassComponentContext);
  if (!data) return null;
  return (
    <svg
      viewBox={`0 0 ${maxDataX + peddingSize} ${maxDataY + peddingSize}`}
      style={{width}}
    >
      {data.map((line) => (
        <Line
          key={line.name}
          hasClassComponent={hasClassComponent}
          line={line}
        />
      ))}
      {labelsX && (
        <ChartAxis
          typeAxis="x"
          labels={labelsX}
          labelPaddingY={maxDataY}
          points={`${0},${maxDataY} ${maxDataX},${maxDataY}`}
        />
      )}
      {labelsY && (
        <ChartAxis
          typeAxis="y"
          labels={labelsY}
          points={`${0},${0} ${0},${maxDataY}`}
          maxDataX={maxDataX}
        />
      )}
    </svg>
  );
};
