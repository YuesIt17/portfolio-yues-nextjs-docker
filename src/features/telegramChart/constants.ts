export const MODULE_NAME = 'telegramChart';
export const CHART_DATA_REDUCER = 'chartData';
export const CHART_LINE_REDUCER = 'chartLine';

export const CHART_WIDTH = 700;
export const SPACE_BETWEEN_LABEL_X = 60;
export const SPACE_BETWEEN_LABEL_Y = 60;
export const FONT_SIZE = 12;
export const PEDDING_CHART_SIZE = 30;
export const CHART_DELTA_MAP_Y = 5;
