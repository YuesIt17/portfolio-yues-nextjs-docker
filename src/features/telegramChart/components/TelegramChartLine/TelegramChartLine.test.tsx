import React from 'react';
import {themeTestRenderer} from '@/utils/tests';
import {TelegramChartLine} from './TelegramChartLine';
import {COLOR_CHART_LINE_JOINED} from '@/share/constants';
import {TChartDataLabel, TChartDataLine} from '@/utils/types';

const dataInput = [
  {
    color: COLOR_CHART_LINE_JOINED,
    name: 'line_one',
    points: '1,0 6,17',
    pointsMap: '1,0 6,3',
    type: 'line',
    isVisible: true,
  },
] as TChartDataLine[];

const labels = [
  {label: '1', coordinate: 10},
  {label: '2', coordinate: 20},
] as TChartDataLabel[];

describe('TelegramChartLine test', () => {
  test('Check render TelegramChartLine component', () => {
    const telegramChartLine = themeTestRenderer(
      <TelegramChartLine
        data={dataInput}
        maxDataX={10}
        maxDataY={10}
        labelsX={labels}
        labelsY={labels}
      />
    ).toJSON();

    expect(telegramChartLine).toMatchSnapshot();
  });
});
