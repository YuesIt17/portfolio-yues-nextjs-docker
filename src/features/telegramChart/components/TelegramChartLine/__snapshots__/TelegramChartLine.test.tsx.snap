// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TelegramChartLine test Check render TelegramChartLine component 1`] = `
<svg
  style={
    Object {
      "width": undefined,
    }
  }
  viewBox="0 0 10 10"
>
  <polyline
    className="css-bqwtyj-line"
    points="1,0 6,17"
  />
  <g>
    <polyline
      className="css-1ied4zn-line"
      points="0,10 10,10"
    />
    <text
      className="css-1nvft8-label"
      x={10}
      y={34}
    >
      1
    </text>
    <text
      className="css-1nvft8-label"
      x={20}
      y={34}
    >
      2
    </text>
  </g>
  <g>
    <polyline
      className="css-1ied4zn-line"
      points="0,0 0,10"
    />
    <g>
      <text
        className="css-1nvft8-label"
        x={0}
        y={0}
      >
        1
      </text>
      <polyline
        className="css-1ied4zn-line"
        points="0,10 10,10"
      />
    </g>
    <g>
      <text
        className="css-1nvft8-label"
        x={0}
        y={8}
      >
        2
      </text>
      <polyline
        className="css-1ied4zn-line"
        points="0,20 10,20"
      />
    </g>
  </g>
</svg>
`;
